src/commands/blockchain-event-listener/blockchain-event-listener-command.js

```diff
@@ -46,37 +46,24 @@ class BlockchainEventListenerCommand extends Command {
 
     async fetchAndHandleBlockchainEvents(blockchainId, repositoryTransaction) {
         const currentBlock = (await this.blockchainEventsService.getBlock(blockchainId)).number;
-
-        const contractEventsData = await Promise.all(
-            Object.values(CONTRACT_EVENTS).map(({ contract, events }) =>
-                this.getContractEvents(blockchainId, contract, currentBlock, events),
-            ),
+        const lastCheckedBlockRecord = await this.repositoryModuleManager.getLastCheckedBlock(
```

Comment:
What happens when the node starts processing events and some of them get handled? The node restarts, we don't update the last checked block, and the node goes through events again even though they were processed.

Before we had logic broken down into steps:

Events get pulled from the chain and we store them, we get all the unprocessed events from the db after that and handle them, as we handle them we mark the as processed